src/tools/schema2base.ts

```diff
@@ -91,6 +98,7 @@ export abstract class Schema2Base {
       }
 
       const graph = new SchemaGraph(particle);
+      const nodes = [];
```

Comment:
Could you add a type for this? Otherwise TypeScript will infer it to be `any[]`.

```
const nodes: NodeAndGenerator[] = [];
```